### ISSUE TITLE
Accuracy definition during run

### ISSUE THREAD

**Issue by mistamun** [tflearn/tflearn#167]
Hello,

I'm adapting the CIFAR-10 example to my own case.
My dataset is around 30k images spread out across 11 classes. Although not equally spread out, as per frequency of each class: [5.4, 13.0, 13.6, 13.7, 10.5, 7.6, 2.0, 5.2, 19.0, 5.1, 5.0] %

From this dataset I randomly picked 15% of the image as my validation set and also set aside another 5% for prediction after training.

```
Training Step: 3336  | total loss: 0.64427
| Adam | epoch: 008 | loss: 0.64427 - acc: 0.7864 | val_loss: 0.73456 - val_acc: 0.7790 -- iter: 26662/26662
Training Step: 3753  | total loss: 0.58631
| Adam | epoch: 009 | loss: 0.58631 - acc: 0.7945 | val_loss: 0.67474 - val_acc: 0.7892 -- iter: 26662/26662
Training Step: 4170  | total loss: 0.54879
| Adam | epoch: 010 | loss: 0.54879 - acc: 0.8210 | val_loss: 0.64050 - val_acc: 0.7977 -- iter: 26662/26662
```
At the end of training
I load the final validation set and run the following code

```
X_eval, Y_eval , image_names = snaps.load_validation_data(keyword='teams', year=args.year)

results = model.predict(X_eval)

hits = 0
print('Prediction, Real Tag, Image')
for index, line in enumerate(results):
    prediction = np.argmax(line)
    reality = Y_eval[index]
    if prediction == reality:
        hits += 1
    print('%s, %s, %s' % (prediction, reality, image_names[index]))

print('Accuracy: %s' % (float(hits)/len(results)))
```

But I never get accuracy which is more than 0.2, 0.3 because the model seems to predict class 1 80% of the time

This is probably a classic case of over-fitting but I can't understand where the high accuracy values come from since the validation set used should be enough to pick up on this over-fitting while the model runs.

So how is that accuracy number really calculated at the end of each epoch? I couldn't find it in the documentation.

Thanks.

**Comment by aymericdamien** [comment]
Hi,

Validation split is performed on a random split of your dataset, and accuracy is computed on that validation set. To verify your results, you can try to prior split your dataset in train/val/test set (after shuffling your data, see `tflearn.data_utils.shuffle` for shuffling at unisson), and verify results by evaluating again the val set when training is done, along with test set.
The issue here may be link with your data, maybe your splitted sets are unbalanced.

**Comment by mistamun** [comment]
This is what I have been using to split the dataset a priory, I think this is okay.
```
val_fraction = 0.15
val_indexes = random.sample(xrange(len(full_data)), int(test_fraction * len(full_data)))
```

Then I'm feeding tensorflow with both, shuffling the train set
```
(X, Y), (X_val, Y_val) = snaps.load_data(keyword='teams', year=args.year)
X, Y = shuffle(X, Y)
.
.
model = tflearn.DNN(network, tensorboard_verbose=0)
model.fit(X, Y, n_epoch=25, shuffle=True, validation_set=(X_val, Y_val),
          show_metric=True, run_id='snaps_teams')
```

I will try verifying with the same set instead of an independent set to see if the result matches.

**Comment by mistamun** [comment]
I have traced the issue to a silly mistake having been mislead by an example of predict I found here, issue  #63 

On that example they were not flatting out the images to be predicted so I wrongly assumed that predict() would digest images in that format for convenience.

In fact it works if the dataset to be predicted is in the same flattened format as the training dataset and evaluation datasets.
